src/DIRAC/Core/Utilities/ReturnValues.py

```diff
@@ -142,3 +151,65 @@ def returnSingleResult(dictRes):
     return S_OK(list(dictRes['Value']['Successful'].values())[0])
   else:
     return S_ERROR("returnSingleResult: Failed and Successful dictionaries are empty")
+
+
+class SErrorException(Exception):
+  """Exception class for use with `convertToReturnValue`"""
+  def __init__(self, result):
+    """Create a new exception return value
+
+    If `result` is a `S_ERROR` return it directly else convert it to an
+    appropriate value using `S_ERROR(result)`.
+
+    :param result: The error to propagate
+    """
+    if not isSError(result):
+      result = S_ERROR(result)
+    self.result = result
+
+
+def returnValueOrRaise(result):
+  """Unwrap an S_OK/S_ERROR response into a value or Exception
+
+  This method assists with using exceptions in DIRAC code by raising
+  :exc:`SErrorException` if `result` is an error. This can then by propogated
+  automatically as an `S_ERROR` by wrapping public facing functions with
+  `@convertToReturnValue`.
+
+  :param result: Result of a DIRAC function which returns `S_OK`/`S_ERROR`
+  :returns: The value associated with the `S_OK` object
+  :raises: If `result["OK"]` is falsey the original exception is re-raised.
```

Comment:
```suggestion
  :raises: If `result["OK"]` is False the original exception is re-raised.
```

Reply:
The current comment is correct as I don't check as it checks `result["OK"]` rather than `result["OK"] is not False`